sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/SelectQuery.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using Azure.DigitalTwins.Core.Queries.QueryBuilders;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Azure.DigitalTwins.Core.QueryBuilder
+{
+    // NOTE -- accessibility internal for now to bypass needing static.
+    /// <summary>
+    /// Custom ADT query builder class that facilitates building queries against an ADT instance.
+    /// </summary>
+    public sealed class SelectQuery : QueryBase
+    {
+        private readonly FromQuery _innerQuery;
+        private readonly AdtQueryBuilder _parent;
+        private SelectClause _clause;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SelectQuery"/> class.
+        /// </summary>
+        internal SelectQuery(AdtQueryBuilder parent, FromQuery select)
+        {
+            _parent = parent;
+            _innerQuery = select;
+        }
+
+        /// <summary>
+        /// Called to add a select clause (and its corresponding argument) to the query.
+        /// </summary>
+        /// <param name="args"> The arguments that define what we select (eg. *). </param>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery Select(params string[] args)
+        {
+            Console.WriteLine(args);
```

Comment:
You can start removing the console logging as you are making progress with the implementation